python_modules/dagit/dagit/webapp/src/graph/SolidNode.tsx

```diff
@@ -75,39 +97,115 @@ export default class SolidNode extends React.Component<ISolidNodeProps> {
   };
 
   renderIO(
-    ComponentClass: StyledComponentClass<any, any, any>,
+    colorKey: string,
     items: Array<SolidNodeFragment_inputs | SolidNodeFragment_outputs>,
     layout: { [inputName: string]: { layout: ILayout } }
   ) {
-    const root = this.props.layout.solid;
-
     return Object.keys(layout).map((key, i) => {
       const { x, y, width, height } = layout[key].layout;
       const input = items.find(o => o.definition.name === key);
 
+      const portInset = 9;
+      const portRadius = 7;
+      const textSize = 15;
+      const namePadding = 7;
+      const typePadding = 4;
+      const name = measureAndClip({
+        text: `${input!.definition.name}: `,
+        textSize,
+        maxWidth: 200
+      });
+      const type = measureAndClip({
+        text: input!.definition.type.name,
+        textSize,
+        maxWidth: 200
+      });
+
+      const baseline = y + height / 2 + textSize / 2;
+
+      let containerWidth =
+        portInset +
+        portRadius * 2 +
+        namePadding +
+        name.width +
+        typePadding +
+        type.width +
+        typePadding +
+        namePadding;
+
+      if (this.props.minified) {
+        containerWidth = 30;
+      }
+
       return (
-        <ComponentClass
-          key={i}
-          style={{ left: x - root.x, top: y - root.y, height: height }}
-        >
-          <Port filled={true} />
+        <g key={i}>
+          <rect
+            x={x}
+            y={y}
+            stroke="#979797"
+            strokeWidth={1}
+            width={width || containerWidth}
+            height={height}
+            fill={PipelineColorScale(colorKey)}
+          />
+          <ellipse
+            cx={x + portInset + portRadius}
+            cy={y + height / 2}
+            rx={portRadius}
+            ry={portRadius}
+            fill="rgba(0, 0, 0, 0.3)"
+            stroke="white"
+          />
           {width == 0 &&
             !this.props.minified && (
-              <InputOutputName>{input!.definition.name}:</InputOutputName>
+              <text
+                x={x + portInset + portRadius * 2 + namePadding}
+                y={baseline}
+                fill="white"
+                style={{ font: `${textSize}px "Source Code Pro", monospace` }}
+              >
+                {name.text}
+              </text>
             )}
           {width == 0 &&
             !this.props.minified && (
-              <TypeName>{input!.definition.type.name}</TypeName>
+              <>
```

Comment:
whats this empty tag thing?

Reply:
Ahh it's a React fragment - just allows you to return an array of elements instead of a single element, without nesting them within a rendered div/span, etc.